pymemcache/client/retrying.py

```diff
@@ -0,0 +1,183 @@
+""" Module containing the RetryingClient wrapper class. """
+
+from collections import Iterable
+from time import sleep
+
+
+def _ensure_tuple_argument(argument_name, argument):
+    """
+    Helper function to ensure the given arguments are tuples of Exceptions (or
+    subclasses), or can at least be converted to such.
+
+    Args:
+      argument_name: str, name of the argument we're checking, only used for
+        raising meaningful exceptions.
+      argument: any, the argument itself.
+
+    Returns:
+      tuple[Exception]: A tuple with the elements from the argument if they are
+        valid.
+
+    Exceptions:
+      ValueError: If the argument was not None, tuple or Iterable.
+      ValueError: If any of the elements of the argument is not a subclass of
+        Exception.
+    """
+
+    argument_tuple = None
+
+    # Ensure the argument is a tuple, or raise an error if we can't convert it.
+    if argument is None:
+        return tuple()
```

Comment:
This could be the default value of `argument_tuple` and the condition block could be reduced to if/elif/else:

Example:
```
if isinstance(argument, tuple):
        argument_tuple = argument
elif isinstance(argument, Iterable) and \
            not isinstance(argument, (str, bytes, bytearray)):
        argument_tuple = tuple(argument)
else:
        raise ValueError("%s was not a tuple or Iterable." % argument_name)
```

Anyway, I don't want to nitpick our current implementation, this code LGTM in the current state.

Reply:
This check can remain. It will be complementary with testing list and tuple. See my next comments.

Reply:
Went ahead and simplified the check as discussed below.